packages/fx-core/src/component/code/utils.ts

```diff
@@ -27,6 +30,140 @@ function capitalizeFirstLetter(raw: string) {
   return raw.charAt(0).toUpperCase() + raw.slice(1);
 }
 
+export async function executeCommand(
+  command: string,
+  projectPath: string,
+  logProvider: LogProvider,
+  ui: DriverContext["ui"],
+  workingDirectory?: string,
+  env?: NodeJS.ProcessEnv,
+  shell?: string,
+  timeout?: number,
+  redirectTo?: string
+): Promise<Result<[string, DotenvOutput], FxError>> {
+  return new Promise(async (resolve, reject) => {
+    let workingDir = workingDirectory || ".";
+    workingDir = path.isAbsolute(workingDir) ? workingDir : path.join(projectPath, workingDir);
+
+    // Drive letter should be uppercase, otherwise when we run webpack in exec, it fails to resolve nested dependencies.
+    if (os.platform() === "win32") {
+      workingDir = capitalizeFirstLetter(path.resolve(workingDir ?? ""));
+    }
+    const defaultShellMap: any = {
+      win32: "powershell",
+      darwin: "bash",
+      linux: "bash",
+    };
+    let run = command;
+    shell = shell ?? defaultShellMap[process.platform];
+    let appendFile: string | undefined = undefined;
+    if (redirectTo) {
+      appendFile = path.isAbsolute(redirectTo) ? redirectTo : path.join(projectPath, redirectTo);
+    }
+    const outputs = parseKeyValueInOutput(command);
+    if (outputs) {
+      resolve(ok(["", outputs]));
+      return;
+    }
+    if (shell === "cmd") {
+      run = `%ComSpec% /D /E:ON /V:OFF /S /C "CALL ${command}"`;
```

Comment:
What's the flags for?

Reply:
%ComSpec% means the location of cmd.
The other part of the arguments can see from [here](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/cmd)